### ISSUE TITLE
Smaller resolution doesn't significantly improve the speed and inference is slower

### ISSUE THREAD

**Issue by ForcewithMe66** [MIC-DKFZ/nnUNet#1682]
When I use nnUNet for 3D segmentation, I meet the following 2 questions.
1. **Smaller resolution doesn't significantly improve the speed.** The original data is `512×512×Z`，the length of dimension `Z` is dynamic, ranging from100 to more than 1000. I use an A100 with 12 CPU cores. Here are the speeds for one epoch with each configuration. `no downsample` indicates I use the original `512×512×Z` data to train. `downsample 4 times` means I manually downsample the data to `128×128×Z/4`. In my opinion, when I downsample 4 times, the data is 4×4×4=64 times smaller. Then even though it will not speed up for 64 times, it should speed up for several times.

```
3d_fullres | no downsample | 120s
3d_lowres | no downsample | 120s
3d_fullres | downsample 4 times| 95s
```

2.  **The inference is so slow**. My training set contains 200 samples and test set contains about 4500 samples. Training one epoch takes 120s(2 mins). But inference on 4500 samples takes more than 30 hours. It's not linear and not even close to linear.

**Comment by constantinulrich** [comment]
Hi, during experiment planning, nnUNet finds the optimized image spacing and patch size for your training. If you reduce the resolution of your raw data, nnUNet adapts the patchsize and network architecture for training. It always tries to make best use out of the GPU memory limit (default 8GB). Therefore different resolutions do not directly influence the training speed. 
For nnU-Net, an epoch is defined by 250 steps -> for one epoch nnUNet sees b*250 patches where b is the batch size. This does not change if you adapt your resolution.

Basically, this also answers also your second question:
During inference, we predict the images using a sliding window approach. Meaning we shift each patch over the whole image using an patch overlap of 50%. Additionally we normally apply a 5 fold ensemble and using testtime augmentation (mirroring) during prediction. To speed up inference you could only train one network with all data (5 times faster) instead of a 5-fold cross validation (5 times faster) and turn of test time augmentation (8 times faster). But of course this comes with a loss of performance.

Best,
Constantin


PS:
Never change the resolution of your preprocessed data!
